python/lsst/rubintv/production/pipelineRunning.py

```diff
@@ -607,6 +626,59 @@ def postProcessVisitSummary(self, quantum: Quantum) -> None:
         except Exception:
             self.log.exception("Failed to populate visit row in ConsDB")
 
+    def postProcessCalcZernikes(self, quantum: Quantum) -> None:
+        """Post-process the Zernike table to send results to ConsDB."""
+        # protect import to stop the whole package depending on ts_wep. If this
+        # becomes a problem we could copy the functions or just accept that RA
+        # needs T&S software.
+        from lsst.ts.wep.utils.zernikeUtils import makeDense
+
+        try:
+            dRef = quantum.outputs["zernikes"][0]
+            zkTable = self.cachingButler.get(dRef)
+        except Exception:
+            self.log.warning(
+                f"Failed to post-process *failed* quantum {quantum}. This is not unexpected"
+                " but still merits a warning due to the failing quantum."
+            )
+            return
+
+        # *ideally* this would be pulled from maxconfig.nollIndices() but
+        # that's not possible here, but a) they never run above 28, and b)
+        # ConsDB only goes out that far, so we'd truncate there anyway, so we
+        # just hardcode it here.
+        MAX_NOLL_INDEX = 28
+
+        # Get the physical rotation from the EFD. Ideally this would be pulled
+        # from the ConsDB, but that's calculated elsewhere in RA, and although
+        # that process is much quicker, using it here is introducing an
+        # unnecessary race condition, so it's better to recalculate it here.
+        visitRecord = dRef.dataId.records["visit"]
+        assert visitRecord is not None, "visitRecord is None, this shouldn't be possible"
+        data = getEfdData(self.efdClient, "lsst.sal.MTRotator.rotation", expRecord=visitRecord)
+        physicalRotation = np.nanmean(data["actualPosition"])
+
+        detector = dRef.dataId.records["detector"]  # not a detector object, but a detector dimension
+        assert detector is not None, "detector is None, this shouldn't be possible"
+        detectorId: int = detector.id  # hence .id rather than .getId()
+
+        zkTable = zkTable[zkTable["label"] == "average"]
```

Comment:
Please add a comment addressing why you are downselecting `zkTable` in this way.

Reply:
We are selecting the average zernike coefficients among all the donuts of that specific detector. `zkTable` provides the estimate for each donut pair, but it suffices to take the average among donut pairs of that detector, for the metrics we want to keep in consDB. If someone wants more details they can query the butler.